### PR TITLE
Fixing integration tests

### PR DESCRIPTION
**Pull request by dangershony**
With this PR the wallet tests are more stable, however more work will go in to this PR

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/TestHelper.cs [hunk 1]
```diff
@@ -43,5 +44,11 @@ public static void TriggerSync(CoreNode node)
             foreach (var connectedPeer in node.FullNode.ConnectionManager.ConnectedPeers)
                 connectedPeer.Behavior<ChainHeadersBehavior>().TrySyncAsync().GetAwaiter().GetResult();
         }
+
+        public static bool IsNodeConnected(CoreNode node)
+        {
+            if (node.FullNode.ConnectionManager.ConnectedPeers.Any()) return true;
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/TestHelper.cs]: You could just `return node.FullNode.ConnectionManager.ConnectedPeers.Any()`

File: src/Stratis.Bitcoin.IntegrationTests/BlockStoreTests.cs [hunk 2]
```diff
@@ -245,6 +245,7 @@ public void BlockStoreCanReorg()
 
                 // remove node 2
                 stratisNodeSync.CreateRPCClient().RemoveNode(stratisNode2.Endpoint);
+                TestHelper.WaitLoop(() => !TestHelper.IsNodeConnected(stratisNode2));
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/BlockStoreTests.cs]: What happens if the node doesn't disconnect at all?
Would it be a good idea to put a one-minute timeout with an error?

File: src/Stratis.Bitcoin.Features.Miner/MiningFeature.cs [hunk 3]
```diff
@@ -241,9 +240,7 @@ public static IFullNodeBuilder AddPowPosMining(this IFullNodeBuilder fullNodeBui
                         services.AddSingleton<IAssemblerFactory, PosAssemblerFactory>();
                         services.AddSingleton<MinerController>();
                         services.AddSingleton<MiningRPCController>();
-                        services.AddSingleton<IWalletManager, WalletManager>();
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Miner/MiningFeature.cs]: I think you need the walletManager for mining, don't you?

File: src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs [hunk 4]
```diff
@@ -60,6 +60,8 @@ public void CanStratisSyncFromCore()
 
                 //Now check if Core connect to stratis
                 stratisNode.CreateRPCClient().RemoveNode(coreNode.Endpoint);
+                TestHelper.WaitLoop(() => coreNode.CreateRPCClient().GetPeersInfo().Length == 0);
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs]: Why not use `TestHelper.IsNodeConnected(coreNode)`?

File: src/Stratis.Bitcoin.IntegrationTests/BlockStoreTests.cs (hunk 2 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.IntegrationTests/BlockStoreTests.cs]: Yeah The method WaitLoop will throw after a timeout

File: src/Stratis.Bitcoin.Features.Miner/MiningFeature.cs (hunk 3 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Miner/MiningFeature.cs]: Well WalletFeature is a dependency so it should already be registered in the DI so this call is actually redundant

File: src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs (hunk 4 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs]: Yeah I tried then realized this is a bitcoin core node.

Comment (bokobza) [reply src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs]: ah right, fair enough 😆

**Pull request merged**